orderapi/src/main/java/fi/hel/verkkokauppa/order/service/order/OrderService.java

```diff
@@ -109,11 +108,26 @@ public OrderAggregateDto getOrderWithItems(final String orderId) {
             List<OrderItem> items = this.orderItemService.findByOrderId(orderId);
             List<OrderItemMeta> metas = this.orderItemMetaService.findByOrderId(orderId);
             orderAggregateDto = orderTransformerUtils.transformToOrderAggregateDto(order, items, metas);
+
+            FlowStep flowStep = getFlowStepsByOrderId(orderId);
+            if (flowStep != null) {
+                orderAggregateDto.setFlowSteps(flowStepMapper.toDto(flowStep));
+            }
         }
 
         return orderAggregateDto;
     }
 
+    public FlowStep getFlowStepsByOrderId(String orderId) {
+        List<FlowStep> flowStep = flowStepRepository.findByOrderId(orderId);
+        if (flowStep.size() > 0) {
+            return flowStep.get(0);
+        }
+
+        log.debug("flow step not found, orderId: " + orderId);
```

Comment:
this can cause a lot of logs because order is fetched multiple times. And I think we dont have this information so often.

Reply:
Yes sounds good, I will remove it.